scripts/InstancingWithModsSystem.js

```diff
@@ -0,0 +1,129 @@
+class InstancingSystem extends System {
+
+    // todo - title it InstancingWithModsSystem
+
+    constructor() {
+        super();
+
+        console.log(this.componentName);
+
+        const this.instanceCount = 5;
+
+        // DataOrientedDesign arrays
+        let this.positions = null;
+        let this.scales = null;
+        let this.rotations = null;
+        let this.meshes = [];
+        let this.geometry = null;
+        console.log('constructed an instancing system');
+    }
+
+    update(deltaTime, frame) {
+        for(const entity of this.registry){
+            switch (entity.components.get('instancing')?.type) {
+                case 'animate':
+                    this.animate(entity)
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    attachedComponent(entity, data) {
+        console.log('in attachedComponent');
+        console.log('entity is:');
+        console.log(entity);
+        console.log('data is:');
+        console.log(data);
+
+        this.createInstances();
+        console.log(entity.object3D);
+        geometry = entity.object3D.geometry;
+        addToScene(geometry);
+        console.log(entity);
+        console.log('done with attachedComponent');
+    }
+
+    createInstances() {
+        // Create buffers for positions, scales, and rotations of instances
+        this.positions = new Float32Array(instanceCount * 3);
+        this.scales = new Float32Array(instanceCount * 3);
+        this.rotations = new Array(instanceCount);
```

Comment:
instanceCount is a property of the class, so you have to call `this.instanceCount`, same for anything declared with `this.` in general.

but also maybe `instanceCount` should be stored in the component itself?

Reply:
√